test/functional/test-viewer.js

```diff
@@ -226,6 +230,100 @@ describe('Viewer', () => {
     expect(viewportEvent).to.not.equal(null);
   });
 
+  describe('replaceUrl', () => {
+    function setUrl(href) {
+      const url = parseUrl(href);
+      windowApi.location.href = url.href;
+      windowApi.location.hash = url.hash;
+    }
+
+    it('should replace URL for the same non-proxy origin', () => {
+      const fragment = '#replaceUrl=http://www.example.com/two&b=1';
+      setUrl('http://www.example.com/one' + fragment);
+      new Viewer(ampdoc);
+      expect(windowApi.history.replaceState).to.be.calledOnce;
+      expect(windowApi.history.replaceState).to.be.calledWith({}, '',
+          'http://www.example.com/two' + fragment);
+      expect(ampdoc.getUrl())
+          .to.equal('http://www.example.com/two' + fragment);
```

Comment:
For that to happen, `&` should be encoded. Basically, the whole fragment is parsed first using query parameter parse, which will end up with two params: `#repalceUrl=http://www.example.com/two` AND `b=1`. Only `replaceUrl` is used for this, so this test demonstrates that `b=1` is ignored.